Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Material/MaterialVersionUpdate.h

```diff
@@ -38,25 +40,69 @@ namespace AZ
                 AZ::Name m_operation;
                 ArgsMap m_argsMap;
 
+                //! Generic structure for arbitrary version update actions that maps nicely to json.
+                //! The operation type is stored as a string under the "op" key.
+                //! All other entries depend on which "op" is being used, see the @ref Action()
+                //! constructor for details.
+                //! Note that we are somewhat hijacking the MaterialPropertyValue class here: not
+                //! every argument will be an actual material property value, some will be generic
+                //! strings like the "op" string or they could be property names for example. But
+                //! when the arg *does* happen to represent a property value it is convenient to
+                //! rely on the associated JsonMaterialPropertyValueSerializer to load the value.
+                //! Since MaterialPropertyValue has the option of holding a string type, we can use
+                //! it for all cases.
+                //! Lastly, note that we don't use an unordered_map to keep json deserialization
+                //! (and the associated unit tests) deterministic.
                 using ActionDefinition = AZStd::map<AZStd::string, MaterialPropertyValue>;
 
                 Action() = default;
-                Action(const AZ::Name& operation, const AZStd::initializer_list<AZStd::pair<AZ::Name, MaterialPropertyValue>>& args);
-                //! The operation type is given as a string under the "op" key, the remaining items define the operation's arguments.
+
+                //! Main constructor for creating a material version update Action from within c++ code.
+                //! @param operation The operation type for this action, supported are
+                //!                   - rename,
+                //!                   - setValue.
+                //! @param args The specific arguments for the chosen operation type:
+                //!              - rename operation: {"from":"oldPropertyName", "to":"newPropertyName"},
+                //!              - setValue operation: {"name":"myInt", "value":100}.
+                //!                In the case where the property's name also has 'rename' version
+                //!                updates, the name to use in 'setValue' is always the final name
+                //!                (even if the 'setValue' occurs at an older version number where
+                //!                that name would not yet have been renamed).
```

Comment:
Is this something we would improve in the future? This kind of comment about nuanced behavior could be come stale if we do; the dev might not realize this comment is here and needs to be updated. Just a thought...